src/corehost/cli/fxr/command_line.cpp

```diff
@@ -0,0 +1,337 @@
+            known_opts.push_back(known_options::additional_deps);
+
+            if (!for_cli_usage)
+            {
+                // Intentionally leave this one out of for_cli_usage since we don't want to show it in command line help (it's deprecated).
+                known_opts.push_back(known_options::roll_forward_on_no_candidate_fx);
+            }
+        }
+
+        return known_opts;
+    }
+
+    const host_option& get_host_option(known_options opt)
+    {
+        int idx = static_cast<int>(opt);
+        assert(0 <= idx && idx < static_cast<int>(known_options::__last));
+        return KnownHostOptions[idx];
+    }
+
+    bool parse_known_args(
+        const int argc,
+        const pal::char_t* argv[],
+        const std::vector<known_options>& known_opts,
+        // Although multimap would provide this functionality the order of kv, values are
+        // not preserved in C++ < C++0x
+        opt_map_t* opts,
+        int* num_args)
+    {
+        int arg_i = *num_args;
+        while (arg_i < argc)
+        {
+            pal::string_t arg = argv[arg_i];
+            pal::string_t arg_lower = pal::to_lower(arg);
+            if (std::find_if(known_opts.begin(), known_opts.end(),
+                [&](const known_options &opt) { return arg_lower == get_host_option(opt).option; })
+                == known_opts.cend())
+            {
+                // Unknown argument.
+                break;
+            }
+
+            // Known argument, so expect one more arg (value) to be present.
+            if (arg_i + 1 >= argc)
+            {
+                return false;
+            }
+
+            trace::verbose(_X("Parsed known arg %s = %s"), arg.c_str(), argv[arg_i + 1]);
+            (*opts)[arg_lower].push_back(argv[arg_i + 1]);
+
+            // Increment for both the option and its value.
+            arg_i += 2;
+        }
+
+        *num_args = arg_i;
+
+        return true;
+    }
+
+    int parse_args(
+        const host_startup_info_t &host_info,
+        int argoff,
+        int argc,
+        const pal::char_t *argv[],
+        bool exec_mode,
+        host_mode_t mode,
+        /*out*/ int *new_argoff,
+        /*out*/ pal::string_t &app_candidate,
+        /*out*/ opt_map_t &opts)
+    {
+        std::vector<known_options> known_opts = get_known_opts(exec_mode, mode);
+
+        // Parse the known arguments if any.
+        int num_parsed = 0;
+        if (!parse_known_args(argc - argoff, &argv[argoff], known_opts, &opts, &num_parsed))
+        {
+            trace::error(_X("Failed to parse supported options or their values:"));
+            for (const auto& opt : known_opts)
+            {
+                const host_option &arg = get_host_option(opt);
+                trace::error(_X("  %-37s  %s"), (arg.option + _X(" ") + arg.argument).c_str(), arg.description.c_str());
+            }
+            return StatusCode::InvalidArgFailure;
+        }
+
+        app_candidate = host_info.app_path;
```

Comment:
Please move this to the if(apphost) case below, it is always overwritten in the else branch.